parsing/parser.mly

```diff
@@ -1701,6 +1701,12 @@ module_type:
             mkmty ~loc:(startpos, $endpos) (Pmty_functor (arg, acc))
           ) mty args
         ) }
+  | args = functor_args
+    MINUSGREATER mty = module_type
+      %prec below_WITH
+      { List.fold_left (fun acc (startpos, arg) ->
+            mkmty ~loc:(startpos, $endpos) (Pmty_functor (arg, acc))
+          ) mty args }
```

Comment:
I think it would be nice to share the `List.fold_left` code (with its non-trivial handling of locations) here and above with an auxiliary function `mk_functor_type`.